test/repositories/GMPBulkSessionRepositorySpec.scala

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package repositories
+
+
+import org.mongodb.scala.bson.BsonDocument
+import org.mongodb.scala.model.{IndexModel, IndexOptions, Indexes}
+import org.scalatest.{BeforeAndAfterEach, OptionValues}
+import org.scalatest.concurrent.{IntegrationPatience, ScalaFutures}
+import org.scalatest.freespec.AnyFreeSpec
+import org.scalatest.matchers.must.Matchers
+import org.scalatestplus.play.guice.GuiceOneAppPerSuite
+import play.api.test.{DefaultAwaitTimeout, FutureAwaits}
+import models.upscan.{UploadStatus, UploadedSuccessfully}
+import models.GMPBulkSession
+import java.util.concurrent.TimeUnit
+
+class GMPBulkSessionRepositorySpec
+  extends AnyFreeSpec with Matchers with ScalaFutures with IntegrationPatience with OptionValues
+    with GuiceOneAppPerSuite with FutureAwaits with DefaultAwaitTimeout with BeforeAndAfterEach {
+
+  val repository: GMPBulkSessionRepository = app.injector.instanceOf[GMPBulkSessionRepository]
+  val id: String = "id"
+  val callBackData: UploadStatus = UploadedSuccessfully("testReference", "testFileName", "testUrl")
+  val emailAddress: String = "testData"
+  val reference: String = "testData"
+  val GMPBulkSession: GMPBulkSession =
+    new GMPBulkSession(id, Some(callBackData), Some(emailAddress), Some(reference))
+
+  override def beforeEach(): Unit = {
+    await(repository.collection.deleteMany(BsonDocument()).toFuture())
+    super.beforeEach()
+  }
+
+  "indexes" - {
+    "are correct" in {
+      repository.indexes.toList.toString() mustBe List(
+        IndexModel(
+          Indexes.ascending("lastModified"),
+          IndexOptions()
+            .name("lastModified")
```

Comment:
could we have a index identical like: lastModifiedIdx

also change in test as well